a11y & template linting fixes

sivakumar-kailasam opened the pull request:
- Fixes color contrast issues with the default colors used in the stylesheets (found using ember-a11y-testing)
- Replaced div with labels
- Added enter action to focus input
- Fixed indentation in the blueprints to be compliant with ember-cli-template-lint
- Make all color variables in stylesheets overridable

chrislopresto commented on app/styles/components/freestyle-palette-item.scss:
```diff
@@ -9,9 +9,9 @@ Hey look... these are `markdown` notes:
-$FreestylePaletteItem-color: #808080;
+$freestylePaletteItemBorderColor: #cecece !default;
+$FreestylePaletteItem-backgroundColor: #fff !default;
+$FreestylePaletteItem-color: #2F4F4F !default;
```
I'm surprised sass-lint didn't break the build here. Should be all lowercase... `#2f4f4f`

chrislopresto commented:
This looks fantastic. Thanks, @sivakumar-kailasam.

How did you choose the new colors? Something close that satisfied the a11y testing?

sivakumar-kailasam commented:
I used this [Color contrast ratio analyser](http://juicystudio.com/services/luminositycontrastratio.php) that was recommended by [Deque](https://dequeuniversity.com/tips/color-contrast).  I kept trying darker shades until I got something I liked 😄

sivakumar-kailasam commented:
@chrislopresto  Fixed the color definition violation that you had mentioned. 

I've updated the plugin in my branch to include their latest changes. But noticed couple of things in the sass lint config in this repo,
- You had ignored the file I had modified in the ignores https://github.com/chrislopresto/ember-freestyle/blob/master/.sass-lint.yml#L10
- The lowercase check applies only for the variable names and not their actual values. Or at least I couldn't find a rule that enforces that :)

chrislopresto commented:
Ha, sass-lint ignore mystery solved. I forget why I ignored that file at the moment. Currently, sass-lint doesn't support disabling a linter on a per-line/block basis, so configuring exceptions will be somewhat clunky until https://github.com/sasstools/sass-lint/pull/677 lands.

FYI, the rule I'm referring to is https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#hexnotation, though I haven't looked at our sass-lint config lately. So it may or may not be configured.

Regardless, I just took a look locally and this looks great. Thanks!

sivakumar-kailasam commented:
Oh I didn't know about sass-lint until I saw it in your repo, thanks for that & this awesome addon! 😄

sivakumar-kailasam commented:
@chrislopresto Can you please release a patch version?

chrislopresto commented:
@sivakumar-kailasam Just released 0.2.13. Cheers.
